crates/net/eth-wire-types/src/status.rs

```diff
@@ -134,6 +135,118 @@ impl Default for Status {
     }
 }
 
+impl PartialEq for Status {
+    fn eq(&self, other: &Self) -> bool {
+        match (self.version, other.version) {
+            (EthVersion::Eth69, EthVersion::Eth69) => {
+                self.chain == other.chain &&
+                    self.blockhash == other.blockhash &&
+                    self.genesis == other.genesis &&
+                    self.forkid == other.forkid
+            }
+            (_, _) => {
+                self.version == other.version &&
+                    self.chain == other.chain &&
+                    self.total_difficulty.unwrap_or(U256::ZERO) ==
+                        self.total_difficulty.unwrap_or(U256::ZERO) &&
+                    self.blockhash == other.blockhash &&
+                    self.genesis == other.genesis &&
+                    self.forkid == other.forkid
+            }
+        }
+    }
+}
+
+impl Encodable for Status {
```

Comment:
I've thought about this a bit, and now that I'm looking at this,

trying to squeeze these into one type is not very ideal.

I'd like to approach this differently by rolling dedicated version types, that should be easier to manage then doing encoding like that.